File: 본문/java/Code_4_06_3.java
import java.util.*;

class Code_4_06_3 {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		final long MOD = 1000000007;
		long a = sc.nextLong();
		long b = sc.nextLong();
		long answer = 1; // a의 0은 1이므로, answer = 1로 초기화
		for (long i = 1; i <= b; i++) {
			answer = (answer * a) % MOD;
		}
		System.out.println(answer);
	}
}

File: 본문/cpp/Code_4_06_3.cpp
#include <iostream>
using namespace std;

const long long mod = 1000000007;
long long a, b, Answer = 1; // a의 0 제곱은 1이므로, Answer=1로 초기화해둡니다.

int main() {
	cin >> a >> b;
	for (int i = 1; i <= b; i++) {
		Answer = (Answer * a) % mod;
	}
	cout << Answer << endl;
	return 0;
}